autoload/neomake/utils.vim

```diff
@@ -320,6 +320,16 @@ let s:unset = {}  " Sentinel.
 " Get a setting by key, based on filetypes, from the buffer or global
 " namespace, defaulting to default.
 function! neomake#utils#GetSetting(key, maker, default, ft, bufnr) abort
+  " Check new-style config.
+  " Add maker and bufnr to context only if g:neomake or b:neomake exist.
+  let context = {'ft': a:ft}
+  if exists('g:neomake') || !empty(getbufvar(a:bufnr, 'neomake'))
+    call extend(context, {'maker': a:maker, 'bufnr': a:bufnr})
+  endif
+  let Ret = neomake#config#get(a:key, g:neomake#config#undefined, context)
```

Comment:
Wouldn't that be an issue for the caller?  I thought that's only a problem if you attempt to call the function in the variable?